frontend/src/pages/TripDetailPage.tsx

```diff
@@ -1,24 +1,60 @@
 import "./TripDetailPage.css";
-import {useParams} from "react-router-dom";
+import {useNavigate, useParams} from "react-router-dom";
+import useTrip from "../hooks/useTrip";
+import LocationMap from "../components/LocationMap";
+import AddLocationForm from "../components/AddLocationForm";
+import Error from "../components/Error";
+import LocationList from "../components/LocationList";
+import Trip from "../types/Trip";
 
+type TripDetailParams = {
+    id: string,
+}
+
+/**
+ * @ToDo title is input field ("border:none") -> can be edited (click on "save" necessary to save changes)
+ */
 export default function TripDetailPage() {
-    // get trip id from url
-    const {id} = useParams<{ id: string }>();
+    const navigate = useNavigate();
+
+    const {id} = useParams<keyof TripDetailParams>() as TripDetailParams;
+    const {trip, notFound, updateTripQuery, deleteTripQuery, removeLocationFromTrip, getShortestPathForTripQuery} = useTrip(id);
 
-    // request trip from API
-    // if request does not exist -> TBD?
+    if (notFound) {
+        return <Error message="Trip not found." link={{text: "Show all trips", to: "/"}}/>
+    }
 
-    // title is input field ("border:none") -> can be edited (click on "save" necessary to save changes)
-    // show delete button -> on click -> request to DELETE Endpoint and navigate bach to DashboardPage
-    // show save button -> PUT API endpoint
-    // show map with location markers
-    // show list of locations
-    //      each item has a "trash icon" (see react-icons) on it -> on click remove location from trip (click on "save" necessary to save changes)
-    // AddLocationForm -> if location is added -> add to trip (click on "save" necessary to save changes)
+    const onDelete = async (trip: Trip) => {
+        await deleteTripQuery(trip);
+        navigate("/");
+    }
 
     return (
         <>
-            <h1>TripDetailPage for Trip {id}</h1>
+            <header className="detail-page-header sticky sticky-top">
```

Comment:
Not sure this will work with the header i built

Reply:
Ja, wahrscheinlich nicht. Multiple stickys gehen laut meiner Recherche nur, wenn sie im selben container liegen. Müssen wir uns mal überlegen, wie wir das machen - vielleicht mit nem offset. Du hast den Header ja fix auf ne Höhe gesetzt oder? 
Soll ich das mal noch einbauen, dass das "sticky" bei -30px top?

Reply:
gefixt